### PR TITLE
Add EclipseLink performance fix as overlay

### PR DESCRIPTION
**Pull request by dazey3**
for #15993

This commit also renames 'io.openliberty.jpa.3.0.thirdparty' to 'io.openliberty.persistence.3.0.thirdparty'

### CODE REVIEW

**Comment by dazey3** [comment]
#build #spawn.fullfat.buckets=com.ibm.ws.jpa_30_fat,com.ibm.ws.jpa_eclipselink_fat

**Review by jhanders34 — APPROVED** [review]

File: dev/io.openliberty.persistence.3.0.thirdparty/bnd.bnd [hunk 1]
```diff
@@ -112,13 +112,12 @@ Include-Resource: \
...
   @${repo;org.eclipse.persistence:org.eclipse.persistence.jpa;${eclFullVersion};EXACT}!/!META-INF/maven/*,\
   @${repo;org.eclipse.persistence:org.eclipse.persistence.jpa.jpql;${eclFullVersion};EXACT}!/!META-INF/maven/*,\
-  @${repo;org.eclipse.persistence:org.eclipse.persistence.jpa.modelgen.processor;${eclFullVersion};EXACT}!/!META-INF/maven/*
+  @${repo;org.eclipse.persistence:org.eclipse.persistence.jpa.modelgen.processor;${eclFullVersion};EXACT}!/!META-INF/maven/*,\
+  org/eclipse/persistence/internal/helper=${bin}/org/eclipse/persistence/internal/helper
 
 publish.wlp.jar.suffix: dev/api/third-party
 
 instrument.classesExcludes: \
-  org/eclipse/persistence/exceptions/i18n/*.class, \
```
[the remark this answers is not in the document]
Comment (dazey3) [reply dev/io.openliberty.persistence.3.0.thirdparty/bnd.bnd]: Perhaps this was an error, but I don't see anything that matches these two excludes. This project has no `.class` files in a directory `org/eclipse/persistence/exceptions/i18n/*`. There are resource files found in `resources/org/eclipse/persistence/exceptions/i18n/`, but they have the file extension ".nlsprops". I could change it to just drop the ".class" portion, but it looked pointless as is.

[the remark this answers is not in the document]
Comment (jhanders34) [reply dev/io.openliberty.persistence.3.0.thirdparty/bnd.bnd]: The nlsprops converts to .class files and we don't want nls class files to be instrumented.  This is common throughout most bundles' exclude lists.

[the remark this answers is not in the document]
Comment (dazey3) [reply dev/io.openliberty.persistence.3.0.thirdparty/bnd.bnd]: Ok, so the matching occurs AFTER the bundle gets built. I undid the change. Thanks for catching this!

**Comment by dazey3** [comment]
#build #spawn.fullfat.buckets=com.ibm.ws.jpa_30_fat,com.ibm.ws.jpa_eclipselink_fat

**Comment by dmuelle** [comment]
Hi @dazey3 - are there mesage updates here that require ID review?

**Comment by dazey3** [comment]
@dmuelle No, the files are just being renamed, not changed in content.

**Comment by tevans78** [comment]
@dazey3 Do you have a link to the upstream issue or pull request? Do you have an idea of when you will be able to remove this overlay?

**Comment by dazey3** [comment]
#run-libby-bot

**Pull request merged**